internal/bzlmod/go_deps.bzl

```diff
@@ -95,49 +221,73 @@ def _go_deps_impl(module_ctx):
                     resolved_version = module_resolutions[path].raw_version,
                 ),
             )
 
-    [
+    for path, module in module_resolutions.items():
         go_repository(
             name = module.repo_name,
             importpath = path,
             sum = module.sum,
             version = "v" + module.raw_version,
-            build_naming_convention = module.build_naming_convention,
-            build_file_proto_mode = module.build_file_proto_mode,
+            build_directives = _get_directives(path, gazelle_overrides),
         )
-        for path, module in module_resolutions.items()
-    ]
-
-    # With transitive dependencies, Gazelle would no longer just have to pass a
-    # single top-level WORKSPACE/MODULE.bazel file, but those of all modules
-    # that use the go_dep tag. Instead, emit a synthetic WORKSPACE file with
-    # Gazelle directives for all of those modules here.
-    directives = {
-        module.repo_name: [
-            "importpath=" + path,
-            "build_naming_convention=" + module.build_naming_convention,
-            "build_file_proto_mode=" + module_tag.build_file_proto_mode,
-        ]
-        for path, module in module_resolutions.items()
-    }
-    _go_repository_directives(
-        name = "bazel_gazelle_go_repository_directives",
-        directives = directives,
+
+    # Create a synthetic WORKSPACE file that lists all Go repositories created
+    # above and contains all the information required by Gazelle's -repo_config
+    # to generate BUILD files for external Go modules. This skips the need to
+    # run generate_repo_config. Only 'importpath' and 'build_naming_convention'
+    # are relevant.
+    _go_repository_config(
+        name = "bazel_gazelle_go_repository_config",
+        importpaths = {
+            module.repo_name: path
+            for path, module in module_resolutions.items()
+        },
+        build_naming_conventions = drop_nones({
+            module.repo_name: get_directive_value(
+                _get_directives(path, gazelle_overrides),
+                "go_naming_convention",
+            )
+            for path, module in module_resolutions.items()
+        }),
     )
 
+    if root_fixups:
+        root_fixups.extend([
+            buildozer_cmd("remove", "build_naming_convention", name = "%go_deps.module"),
+            buildozer_cmd("remove", "build_file_proto_mode", name = "%go_deps.module"),
+        ])
+
+        print("""
+
+The 'build_naming_convention' and 'build_proto_file_mode' attributes of \
+go_deps.module have been replaced with the more general go_deps.gazelle_override \
+tag and will be removed in the next release of rules_go.
+
+To migrate manually, add a gazelle_override tag for all Go module paths that set \
+one of these attributes and add "gazelle:go_naming_convention <value>" (for \
+build_naming_convention) or "gazelle:proto <value>" (for build_proto_file_mode) \
+to its 'directives' attribute.
```

Comment:
```suggestion
to its "directives" attribute.
```

Reply:
I am using single quotes around attributes in this comment (also in the first paragraph) to distinguish them from Starlark strings rendered in double quotes.